### PR TITLE
Add AWS arguments to `generate-script` commands

### LINKED ISSUE #690: Add AWS support to `gh gei generate-script`
We added support for AWS as intermediary storage with https://github.com/github/gh-gei/issues/594 however we only added it to the `migrate-repo` command at the time. We should also add the same arguments to `generate-script`.

### PR DESCRIPTION
**Pull request by tambling**
Closes #690

The PR mainly adds `--aws-bucket-name` to `gei generate-script` command in order to support AWS S3 archive upload as an alternative to Azure Storage. Additionally the `--azure-storage-connection-string` option was removed from the `gei generate-script` command since it is considered a security risk to include any secrets in the generated script. 

- [x] Did you write/update appropriate tests
- [x] Release notes updated (if appropriate)
- [x] Appropriate logging output
- [x] Issue linked
- [x] Docs updated (or issue created)

### CODE REVIEW

**Comment by timrogers** [comment]
@tambling 👋🏻 I know that there is at least one customer blocked by this. Do you think you'll have a chance to move this PR forward this week? Or would it be worth handing it off to someone else?

**Comment by timrogers** [comment]
@ArinGhazarian is working on this now 🙏🏻 

Fixes https://github.com/github/gh-gei/issues/690.

File: src/gei/Handlers/MigrateRepoCommandHandler.cs [hunk 1]
```diff
@@ -368,5 +370,53 @@ private void ValidateOptions(MigrateRepoCommandArgs args)
...
+        // GHES migration path
+        if (args.GhesApiUrl.HasValue())
+        {
+            var shouldUseAzureStorage = args.AzureStorageConnectionString.HasValue() || _environmentVariableProvider.AzureStorageConnectionString().HasValue();
+            var shouldUseAwsS3 = args.AwsBucketName.HasValue();
+
+            if (!shouldUseAzureStorage && !shouldUseAwsS3)
+            {
+                throw new OctoshiftCliException(
+                    "Eitehr Azure storage connection (--azure-storage-connection-string or AZURE_STORAGE_CONNECTION_STRING env. variable) or " +
```
Comment (dylan-smith) [inline src/gei/Handlers/MigrateRepoCommandHandler.cs:383]:
```suggestion
                    "Either Azure storage connection (--azure-storage-connection-string or AZURE_STORAGE_CONNECTION_STRING env. variable) or " +
```

File: src/gei/Handlers/MigrateRepoCommandHandler.cs [hunk 2]
```diff
@@ -368,5 +370,53 @@ private void ValidateOptions(MigrateRepoCommandArgs args)
...
+                    "AWS S3 connection (--aws-bucket-name, --aws-access-key (or AWS_ACCESS_KEY evn. variable), --aws-secret-key (or AWS_SECRET_Key env.variable)) " +
```
Comment (dylan-smith) [inline src/gei/Handlers/MigrateRepoCommandHandler.cs:384]:
```suggestion
                    "AWS S3 connection (--aws-bucket-name, --aws-access-key (or AWS_ACCESS_KEY env. variable), --aws-secret-key (or AWS_SECRET_Key env.variable)) " +
```

**Comment by dylan-smith** [comment]
Can you add a release note and link the issue

File: src/gei/Handlers/MigrateRepoCommandHandler.cs (hunk 1 above)
Comment (ArinGhazarian) [reply src/gei/Handlers/MigrateRepoCommandHandler.cs:383]: OMG! typos :(

**Review by dylan-smith — APPROVED** [review]

**Pull request merged**